core/src/test/scala/org/apache/spark/network/buffer/LargeByteBufferInputStreamSuite.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.network.buffer
+
+import java.io.{FileInputStream, FileOutputStream, OutputStream, File}
+import java.nio.ByteBuffer
+import java.nio.channels.FileChannel.MapMode
+
+import org.junit.Assert._
+import org.scalatest.{FunSuite, Matchers}
+
+import org.apache.spark.network.buffer.{LargeByteBuffer, LargeByteBufferHelper, WrappedLargeByteBuffer}
+
+class LargeByteBufferInputStreamSuite extends FunSuite with Matchers {
+
+  test("read from large mapped file") {
+    val testFile = File.createTempFile("large-buffer-input-stream-test",".bin")
+    testFile.deleteOnExit()
```

Comment:
IIUC, this test is creating a humongous test file (> 2GB)... two things:

- how long does it take for the test to run?
- this file should probably be explicitly deleted in a `finally` block in this test (or using `BeforeAndAfter`), instead of relying on `deleteOnExit`. Just because I'd be wary of leaking 2GB files.